src/Symfony/Component/FeatureToggle/FeatureChecker.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureToggle;
+
+final class FeatureChecker implements FeatureCheckerInterface
+{
+    public function __construct(
+        private readonly FeatureCollection $features,
+        private readonly bool $whenNotFound,
```

Comment:
Shouldn't the feature you ask for always exist? :thinking: What's the point of calling it with a feature that does not exist?

Reply:
You might have missed some checks and you don't want to display an error. You want a global fallback value. Usually it will be `false` but it might be `true` for some reason.

Reply:
Might be worth doing a command similar to `translation:extract` to find all feature flag usages in the code, which are not covered by config. Not sure how it will work with external provider though.

Reply:
that's the thing. External provider can be configured on the fly to provide a flag or remove it. Having a not found feature flag is not an invalid state in these cases. But a debug command to list where each feature is being checked yes that would be awesome. Before doing that let's see if the component will be accepted or not as it is or if it need a major rework.